src/FingerWhoIs.Lib/Native/NativeApi.cs

```diff
@@ -139,7 +139,14 @@ internal static unsafe NativeCall<List<int>> Indentify(ReadOnlySpan<byte> fmd,Fm
             Span<int> fmdSizes = fmds.Length <= 128 ? stackalloc int[fmds.Length] : new int[fmds.Length];
             for (int i = 0; i < fmds.Length; i++)
             {
-                fmdSizes[i] = fmds[i].Length;
+                if (fmds[i] == null)
+                {
+                    fmdSizes[i] = 0;
+                }
+                else
+                {
+                    fmdSizes[i] = fmds[i].Length;
+                }
             }
```

Comment:
Can this logic be moved to the FingerEngine Class instead ? the NativeApi serves as a point for marshalling and native calls only, this kind of logic should be done before a call is placed here

Reply:
Okay

Reply:
Updated